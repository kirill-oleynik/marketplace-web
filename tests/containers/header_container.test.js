import React from 'react';

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ConnectedHeaderContainer, { HeaderContainer } from 'app/containers/header_container';

const fakeStore = (state = {}) => ({
  subscribe() {},
  dispatch() {},
  getState() { return state; }
});

const currentUser = {
  id: Symbol('userId')
};

const application = {
  id: 1,
  title: 'Test1'
};

const category = {
  id: 2,
  title: 'Test2'
};

const applications = {
  ids: [1],
  byId: {
    1: application
  }
};

const categories = {
  byId: {
    2: category
  }
};

const search = {
  categoryIds: [2],
  applicationIds: [1]
};

describe('#render', () => {
  test('it renders correctly', () => {
    const store = fakeStore({
      search,
      categories,
      applications,
      currentUser
    });

    const component = shallow(
      <ConnectedHeaderContainer />,
      { context: { store } }
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#passedProps', () => {
  test('it passess props to component correctly', () => {
    const store = fakeStore({
      search,
      categories,
      applications,
      currentUser
    });

    const component = shallow(
      <ConnectedHeaderContainer />,
      { context: { store } }
    );

    const props = component.props();

    expect(props.currentUser).toEqual(currentUser);
    expect(props.searchCategories).toEqual([category]);
    expect(props.searchApplications).toEqual([application]);
    expect(props.signOut).toBeInstanceOf(Function);
    expect(props.searchFetch).toBeInstanceOf(Function);
    expect(props.toggleProfileModal).toBeInstanceOf(Function);
    expect(props.toggleSubmitApplicationModal).toBeInstanceOf(Function);
  });
});

describe('#openProfileModal', () => {
  test('it calls toggleProfileModal', () => {
    const toggleProfileModalSpy = jest.fn();

    const component = shallow(
      <HeaderContainer
        currentUser={{}}
        toggleProfileModal={toggleProfileModalSpy}
        toggleSubmitApplicationModal={() => {}}
      />
    );

    component.instance().openProfileModal();

    expect(toggleProfileModalSpy).toHaveBeenCalledWith(true);

    toggleProfileModalSpy.mockReset();
    toggleProfileModalSpy.mockRestore();
  });
});

describe('#openSubmitApplicationModal', () => {
  test('it calls toggleSubmitApplicationModal', () => {
    const toggleSubmitApplicationModalSpy = jest.fn();

    const component = shallow(
      <HeaderContainer
        currentUser={{}}
        toggleProfileModal={() => {}}
        toggleSubmitApplicationModal={toggleSubmitApplicationModalSpy}
      />
    );

    component.instance().openSubmitApplicationModal();

    expect(toggleSubmitApplicationModalSpy).toHaveBeenCalledWith(true);

    toggleSubmitApplicationModalSpy.mockReset();
    toggleSubmitApplicationModalSpy.mockRestore();
  });
});

describe('#handleMenuClick', () => {
  test('it toggles isMenuOpen state', () => {
    const component = shallow(
      <HeaderContainer
        currentUser={{}}
        toggleProfileModal={() => {}}
        toggleSubmitApplicationModal={() => {}}
      />
    );

    expect(component.state().isMenuOpen).toEqual(false);
    component.instance().handleMenuClick();
    expect(component.state().isMenuOpen).toEqual(true);
    component.instance().handleMenuClick();
    expect(component.state().isMenuOpen).toEqual(false);
  });
});
