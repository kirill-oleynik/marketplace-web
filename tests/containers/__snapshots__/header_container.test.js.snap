// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<div>
  <header
    className="main-header"
  >
    <div
      className="container"
    >
      <div
        className="main-header__content"
      >
        <div
          className="hidden-sm-up"
        >
          <a
            className="main-header__btn-menu icon icon-menu in-white"
            href="/#"
            onClick={[Function]}
          />
        </div>
        <a
          className="page-main-logo"
          href="/"
        />
        <div
          className="hidden-xs-down"
        >
          <button
            className="main-btn main-btn--transparent main-btn--md"
            type="button"
          />
          <a
            className="main-header__link"
            href="/log_in"
            onClick={[Function]}
          />
          <button
            className="main-btn main-btn--white main-btn--md"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="hidden-sm-up"
        >
          <a
            className="main-header__search-btn icon icon-search"
            href="/"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
